subsys/bluetooth/host/monitor.c

```diff
@@ -241,6 +216,108 @@ extern void __stdout_hook_install(int (*fn)(int));
 #define CONFIG_BT_MONITOR_ON_DEV_NAME CONFIG_UART_CONSOLE_ON_DEV_NAME
 #endif
 
+struct monitor_log_ctx {
+	size_t total_len;
+	char msg[MONITOR_MSG_MAX];
+};
+
+static int monitor_log_out(u8_t *data, size_t length, void *user_data)
+{
+	struct monitor_log_ctx *ctx = user_data;
+	size_t i;
+
+	for (i = 0; i < length && ctx->total_len < sizeof(ctx->msg); i++) {
+		/* FIXME: There should be a way to request the log subsystem
+		 * not to add these to the end of messages.
+		 */
+		if (data[i] == '\r' || data[i] == '\n') {
+			break;
```

Comment:
@nordic-krch could we get a flag for the log subsystem so that I don't need to do this kind of truncation?